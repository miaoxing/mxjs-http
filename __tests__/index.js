import api from '..';
import {createPromise, bootstrap} from '@mxjs/test';
import $ from 'miaoxing';

bootstrap();

describe('test', () => {
  test('get', async () => {
    const promise = createPromise();
    $.http = jest.fn().mockImplementationOnce(() => promise.resolve({
      ret: {
        data: {
          foo: 3,
          bar: 4,
        },
      },
    }));

    const ret = await api.get('abc', {loading: true});
    expect(ret).toMatchSnapshot();

    expect($.http).toHaveBeenCalledTimes(1);
    expect($.http).toMatchSnapshot();
  });
});

