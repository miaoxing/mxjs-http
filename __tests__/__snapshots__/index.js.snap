// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test get 1`] = `
Object {
  "ret": Object {
    "data": Object {
      "bar": 4,
      "foo": 3,
    },
  },
}
`;

exports[`test get 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "loading": true,
        "method": "get",
        "url": "/api/abc",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
  ],
}
`;
